### PR TITLE
fix the tanh and fix learning rate due to default dtype

### PR DESCRIPTION
**Pull request by ForFishes**
### PR types

Bug fixes 
### PR changes

APIs
### Describe

fix the `tanh` and `adam` for default dtype

### CODE REVIEW

File: python/paddle/optimizer/optimizer.py [hunk 1]
```diff
@@ -336,7 +338,8 @@ def _create_global_learning_rate(self):
                 name=unique_name.generate("learning_rate"),
                 shape=[1],
                 value=float(self._learning_rate),
-                dtype='float32' if self._dtype is None else self._dtype,
+                dtype=paddle.get_default_dtype()
+                if self._dtype is None else self._dtype,
```
Comment (jzhang533) [inline python/paddle/optimizer/optimizer.py:342]: 这个改动会影响到哪些API呀？
相应的地方的文档，也需要看一下，是否要修改。

**Review by jzhang533 — APPROVED** [review]
lgtm
